meson.build

```diff
@@ -1020,9 +1020,9 @@ foreach target : ['default-target'] + targets
       if host_cpu_family == 'riscv'
         # ld.lld before version 15 did not support linker relaxations, disable
         # them if we are using an older version.
-        # There is no `cc.get_linker_version()` function, so we detect ld.lld
-        # version 15 by checking for a newly added linker flag.
-        if not cc.has_link_argument('--package-metadata=1')
```

Comment:
I am quite surprised this is not working, I tested LLD 14 and 15. The meson configure logs would identify why it is failing. I wonder if it has been removed in a later version? It is a somewhat fragile test, but generally ld.lld does not remove flags.

I'm not particularly keen on assuming clang+lld are bundled together since this prevents you from using another compiler with ld.lld as the linker. Would a regex match on the linker '--version' output be acceptable instead?

Reply:
Yeah, I was a bit surprised it stopped working as well. It could be that something has changed in how meson is doing the test or in how the clang arguments are managed within the cross file.

Using an explicit linker version check with a regex seems like a slightly better fix; feel free to cook something up.

Reply:
I am guessing that retrieving the lld version is not completely straightforward:

```
$ lld --version
lld is a generic driver.
Invoke ld.lld (Unix), ld64.lld (macOS), lld-link (Windows), wasm-ld (WebAssembly) instead

$ ld.lld --version
Ubuntu LLD 14.0.0 (compatible with GNU linkers)
```
The first question is what the lld linker is called, and then you need to ignore rubbish like the "Ubuntu" prefix and the "compatible with" suffix, which may change and do not have much to do with the version number we are looking for.

I would normally use a Bash script I have developed for this purpose:

https://github.com/rdiez/Tools/tree/master/CheckVersion

However, that tool is geared towards shell scripts, and it's AGPL 3.0. But Meson can probably capture the output of a process and retrieve the version components using a regular expression too.

Reply:
FreeBSD does something similar in bsd.linker.mk to extract the version. But it seems like the real problem was the missing `-Wl` prefix that happened to work before but no longer seems to work with recent clang/meson.